### ISSUE TITLE
[A11Y] Add missing attributes to modal

### ISSUE THREAD

**Issue by davwheat** [flarum/framework#3516]
**Note:** this could be performed during a migration to `<dialog>` HTML elements.

**Comment by davwheat** [comment]
Added this and the PR to the milestone as the PR's main focus isn't this issue itself.
